lms/djangoapps/courseware/field_overrides.py

```diff
@@ -24,7 +24,7 @@
 from xmodule.modulestore.inheritance import InheritanceMixin
 
 NOTSET = object()
-ENABLED_OVERRIDE_PROVIDERS_KEY = "courseware.field_overrides.enabled_providers"
+ENABLED_OVERRIDE_PROVIDERS_KEY = "courseware.field_overrides.enabled_providers.{course_id}"
```

Comment:
Previously the providers cache was global, populated the first time any block was wrapped by a `FieldOverrideProvider`, and never invalidated. This led to bugs when there were two or more courses with different field overrides configured where courses could receive the wrong override providers due to the caching behavior. I suspect that this bug was never encountered before simply because we haven't had that many courses using providers which aren't always enabled.